app.py

```diff
@@ -0,0 +1,151 @@
+from io import BytesIO
+import base64
+from js import fetchData
+
+from PIL import Image, ImageDraw, ImageFont
+
+
+# since base excuse image is fixed, this values are also constant
+IMAGE_WIDTH = 413
+# Y text coordinates
+WHO_TEXT_Y = 12
+LEGIT_TEXT_Y = 38
+WHY_TEXT_Y = 85
+WHAT_TEXT_Y = 222
+
+
+async def _get_text_font(size: int) -> ImageFont:
+    """
+    Loads font and sets font size for text on image
+
+    :param size: font size
+
+    :returns: ImageFont object with desired font size set
+    """
+    data = await fetchData("/xkcd-script.ttf")
+    buffer = BytesIO(bytes(data))
+    return ImageFont.truetype(buffer, size)
+
+async def _get_blank_image() -> Image:
+    """
+    Loads font and sets font size for text on image
+
+    :param size: font size
+
+    :returns: ImageFont object with desired font size set
+    """
+    data = await fetchData("/blank_excuse.png")
+    buffer = BytesIO(bytes(data))
+    return Image.open(buffer)
+
+def _check_user_input_not_empty(
+    errors: list, text: str, error_code: int) -> list:
+    """
+    Checks if user input size can actually fit in image.
+    If not, add an error to existing list of errors.
+
+    :param errors: list of errors
+    :param text: user's input
+    :param error_code: internal error code
+
+    :returns: list of errors
+    """
+    if not text or text.strip() == '':
+        errors.append({
+            'code': error_code,
+            'message': 'This field is required.'
+        })
+    return errors
+
+def _check_user_input_size(errors: list, max_width: float, text: str,
+    text_font: ImageFont, error_code: int) -> list:
+    """
+    Checks if user input size can actually fit in image.
+    If not, add an error to existing list of errors.
+
+    :param errors: list of errors
+    :param max_width: max size of text
+    :param text: user's input
+    :param text_font: font to be displayed on image
+    :param error_code: internal error code
+
+    :returns: list of errors
+    """
+    if text_font.getlength(text) > max_width:
+        errors.append({
+            'code': error_code,
+            'message': 'Text too long.'
+        })
+    return errors
+
+
+
+async def get_excuse_image(who: str, why: str, what: str):
+    """
+    Load excuse template and write on it.
+    If there are errors (non-existant text, some text too long),
+    return list of errors.
+
+    :param who: who's excuse
+    :param why: what is the excuse
+    :param what: what are they saying
+
+    :returns: pillow Image object with excuse written on it
+    """
+    errors = []
+
+    errors = _check_user_input_not_empty(errors, who, 1010)
+    errors = _check_user_input_not_empty(errors, why, 1020)
+    errors = _check_user_input_not_empty(errors, what, 1030)
+
+    who = 'The #1  {} excuse'.format(who).upper()
+    legit = 'for legitimately slacking off:'.upper()
+    why = '"{}."'.format(why)
```

Comment:
Please add `.upper()` to `why` and `what`. In the [original code](https://github.com/mislavcimpersak/xkcd-excuse-generator/blob/master/app.py#L243) a separate function did that, but think we can just do it here.

Reply:
Done! Force pushed the changes so everything is still in one commit